File: lib/src/clean/domain/entities/unico.config.dart
class UnicoConfig {
  late String _getProjectNumber;
  late String _getProjectId;
  late String _getMobileSdkAppId;
  late String _getBundleIdentifier;
  late String _getHostInfo;
  late String _getHostKey;

  UnicoConfig({
    required String getProjectNumber,
    required String getProjectId,
    required String getMobileSdkAppId,
    required String getBundleIdentifier,
    required String getHostInfo,
    required String getHostKey,
  }){
    _getProjectNumber = getProjectNumber;
    _getProjectId = getProjectId;
    _getMobileSdkAppId = getMobileSdkAppId;
    _getBundleIdentifier = getBundleIdentifier;
    _getHostInfo = getHostInfo;
    _getHostKey = getHostKey;
  }

  Map<dynamic, dynamic> get getCommonMap {
    return <dynamic, dynamic>{
      'project_number': _getProjectNumber,
      'project_id': _getProjectId,
      'mobile_sdk_app_id': _getMobileSdkAppId,
      'bundle_identifier': _getBundleIdentifier,
      'host_info': _getHostInfo,
      'host_Key': _getHostKey
    };
  }
}

File: ios/Classes/config/UnicoConfig.swift
//
//  UnicoConfig.swift
//  Pods
//
//  Created by Lucas Diniz da Silva on 27/05/22.
//

import Foundation


class UnicoConfig : NSObject,  AcessoBioConfigDataSource {
    private var argument: Dictionary<String, Any>?
    
    init(argument: Dictionary<String, Any>){
        self.argument = argument[MethodConstants.UNICO_CONFIG_IOS.rawValue] as? Dictionary<String, Any> ?? nil
    }
    
    func getProjectNumber() -> String {
        return self.argument?[MethodConstants.PROJECT_NUMBER.rawValue] as? String ?? ""
    }
    
    func getProjectId() -> String {
        return self.argument?[MethodConstants.PROJECT_ID.rawValue] as? String ?? ""
    }
    
    func getMobileSdkAppId() -> String {
        return self.argument?[MethodConstants.MOBILE_SDK_APP_ID.rawValue] as? String ?? ""
    }
    
    func getBundleIdentifier() -> String {
        return self.argument?[MethodConstants.BUNDLE_IDENTIFIER.rawValue] as? String ?? ""
    }
    
    func getHostInfo() -> String {
        return self.argument?[MethodConstants.HOST_INFO.rawValue] as? String ?? ""
    }
    
    func getHostKey() -> String {
        return self.argument?[MethodConstants.HOST_KEY.rawValue] as? String ?? ""
    }
}
